diff --git a/include/netlink/msg.h b/include/netlink/msg.h
index f034691..b9c4e0e 100644
--- a/include/netlink/msg.h
+++ b/include/netlink/msg.h
@@ -12,6 +12,15 @@
 #ifndef NETLINK_MSG_H_
 #define NETLINK_MSG_H_
 
+/*struct ucred {
+       pid_t pid;
+       uid_t uid;
+       gid_t gid;
+};*/
+
+#define _GNU_SOURCE
+#include <sys/socket.h>  /* struct ucred is here in musl */
+#include <sys/types.h>   /* pid_t, uid_t, gid_t */
 #include <netlink/netlink.h>
 #include <netlink/object.h>
 
